OsmAnd-java/src/main/java/net/osmand/GPXUtilities.java

```diff
@@ -1657,8 +1660,14 @@ public int getNonEmptySegmentsCount() {
 		}
 
 		public QuadRect getRect() {
-			double left = 0, right = 0;
-			double top = 0, bottom = 0;
+			return getRectAroundPoint(0, 0);
+		}
+
+		public QuadRect getRectAroundPoint(double lat, double lon) {
+			return getBounds(lon, lat, lon, lat);
+		}
+
+		public QuadRect getBounds(double left, double bottom, double right, double top) {
 			for (Track track : tracks) {
 				for (TrkSegment segment : track.segments) {
 					for (WptPt p : segment.points) {
```

Comment:
if (left == 0 && right == 0) { is not correct here ! cause 0 is not defined in this method at all